pkg/gardenlet/controller/networkpolicy/add_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkpolicy_test
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	. "github.com/gardener/gardener/pkg/gardenlet/controller/networkpolicy"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+)
+
+var _ = Describe("Add", func() {
+	var (
+		reconciler *Reconciler
+	)
+
+	BeforeEach(func() {
+		reconciler = &Reconciler{
+			GardenNamespace: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: v1beta1constants.GardenNamespace,
+				},
+			},
+			IstioSystemNamespace: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: v1beta1constants.IstioSystemNamespace,
+				},
+			},
+		}
+	})
+
+	Describe("#MapToNamespaces", func() {
+		var (
+			ctrl              *gomock.Controller
+			mockRuntimeClient *mockclient.MockClient
+			ctx               = context.TODO()
+			log               = logr.Discard()
+			namespaces        []corev1.Namespace
+		)
+
+		BeforeEach(func() {
+			namespaces = []corev1.Namespace{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   "shoot-foo",
+						Labels: map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   "shoot-bar",
+						Labels: map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot},
+					},
+				},
+			}
+		})
+
+		JustBeforeEach(func() {
+			ctrl = gomock.NewController(GinkgoT())
+			mockRuntimeClient = mockclient.NewMockClient(ctrl)
```

Comment:
Do not use mock client for new tests, use fake client instead (ref https://github.com/gardener/gardener/blob/master/docs/development/testing.md#writing-unit-tests)